smartsim/_core/entrypoints/telemetrymonitor.py

```diff
@@ -0,0 +1,677 @@
+            run_entities["ensemble"],
+        )
+        the_runs.append(run)
+
+    return the_runs
+
+
+def load_manifest(file_path: str) -> t.Optional[RuntimeManifest]:
+    """Load a persisted manifest and return the content"""
+    manifest_dict: t.Optional[t.Dict[str, t.Any]] = None
+    try_count = 1
+
+    while manifest_dict is None and try_count < 6:
+        source = pathlib.Path(file_path)
+        source = source.resolve()
+
+        try:
+            if text := source.read_text(encoding="utf-8").strip():
+                manifest_dict = json.loads(text)
+        except json.JSONDecodeError as ex:
+            print(f"Error loading manifest: {ex}")
+            # hack/fix: handle issues reading file before it is fully written
+            time.sleep(0.5 * try_count)
+        finally:
+            try_count += 1
+
+    if not manifest_dict:
+        return None
+
+    exp = manifest_dict.get("experiment", None)
+    if not exp:
+        raise ValueError("Manifest missing required experiment")
+
+    runs = manifest_dict.get("runs", None)
+    if runs is None:
+        raise ValueError("Manifest missing required runs")
+
+    exp_dir = pathlib.Path(exp["path"])
+    runs = hydrate_runs(runs, exp_dir)
+
+    manifest = RuntimeManifest(
+        name=exp["name"],
+        path=exp_dir,
+        launcher=exp["launcher"],
+        runs=runs,
+    )
+    return manifest
+
+
+def track_event(
+    timestamp: int,
+    task_id: t.Union[int, str],
+    step_id: str,
+    etype: str,
+    action: _EventClass,
+    status_dir: pathlib.Path,
+    logger: logging.Logger,
+    detail: str = "",
+    return_code: t.Optional[int] = None,
+) -> None:
+    """
+    Persist a tracking event for an entity
+    """
+    tgt_path = status_dir / f"{action}.json"
+    tgt_path.parent.mkdir(parents=True, exist_ok=True)
+
+    try:
+        task_id = int(task_id)
+    except ValueError:
+        pass
+
+    entity_dict = {
+        "timestamp": timestamp,
+        "job_id": task_id,
+        "step_id": step_id,
+        "type": etype,
+        "action": action,
+    }
+
+    if detail is not None:
+        entity_dict["detail"] = detail
+
+    if return_code is not None:
+        entity_dict["return_code"] = return_code
+
+    try:
+        if not tgt_path.exists():
+            # Don't overwrite existing tracking files
+            bytes_written = tgt_path.write_text(json.dumps(entity_dict, indent=2))
+            if bytes_written < 1:
+                logger.warning("event tracking failed to write tracking file.")
+    except Exception:
+        logger.error("Unable to write tracking file.", exc_info=True)
+
+
+def faux_return_code(step_info: StepInfo) -> t.Optional[int]:
+    """Create a faux return code for a task run by the WLM. Must not be
+    called with non-terminal statuses or results may be confusing"""
+    if step_info.status not in TERMINAL_STATUSES:
+        return None
+
+    if step_info.status == STATUS_COMPLETED:
+        return 0
```

Comment:
I think we should also be able to change these to the `os.EX_` numbers?

Reply:
Good call!